src/metrics.ts

```diff
@@ -0,0 +1,143 @@
+import { Weapon } from "./weapon";
+
+export enum MetricPath {
+  WINDUP_HORIZONTAL = "attacks.horizontal.light.windup",
+  WINDUP_OVERHEAD = "attacks.overhead.light.windup",
+  WINDUP_STAB = "attacks.stab.light.windup",
+  WINDUP_SPECIAL = "specialAttack.windup",
+
+  RANGE_HORIZONTAL = "attacks.horizontal.range",
+  RANGE_ALT_HORIZONTAL = "attacks.horizontal.altRange",
+  RANGE_OVERHEAD = "attacks.overhead.range",
+  RANGE_ALT_OVERHEAD = "attacks.overhead.altRange",
+  RANGE_STAB = "attacks.stab.range",
+  RANGE_ALT_STAB = "attacks.stab.altRange",
+  // RANGE_SPECIAL = "specialAttack.range", TODO
+
+  DAMAGE_HORIZONTAL_LIGHT = "attacks.horizontal.light.damage",
+  DAMAGE_HORIZONTAL_HEAVY = "attacks.horizontal.heavy.damage",
+  DAMAGE_OVERHEAD_LIGHT = "attacks.overhead.light.damage",
+  DAMAGE_OVERHEAD_HEAVY = "attacks.overhead.heavy.damage",
+  DAMAGE_STAB_LIGHT = "attacks.stab.light.damage",
+  DAMAGE_STAB_HEAVY = "attacks.stab.heavy.damage",
+  DAMAGE_SPECIAL = "specialAttack.damage"
+}
```

Comment:
As the data model evolves, this can be updated to add new potential metrics, and to move fields around